docker-compose/tests/identification.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash -e
+#Script to test the integration between the dashboard and the identification module in the mF2C Project
+#
+
+printf '\e[0;33m %-15s \e[0m \n' "The installation of the jq package is a prerequisite to run the test"
+printf '\e[0;33m %-15s \e[0m Starting...\n' [Identification-Test]
+
+function log
+{
+    text="$2"
+    if [[ $1 == "OK" ]]
+    then
+        printf '\e[0;33m %-15s \e[32m SUCCESS:\e[0m %s \n' [Identification-Test] "$text"
+    else
+        printf '\e[0;33m %-15s \e[0;31m FAILED:\e[0m %s \n' [Identification-Test] "$text"
+    fi
+}
+
+endpoint="http://localhost:46060/api/v1/resource-management/identification/requestID"
```

Comment:
This endpoint is failing with the mF2C stack as Identification doesn't have any port published in host. Please attach to the container and perform the command.

```bash
mf2c@atlas-node12:~/test-identification/mF2C/docker-compose$ curl -GET ${endpoint}
curl: (7) Failed to connect to localhost port 46060: Connection refused
```

Reply:
Here you have an example: https://github.com/mF2C/mF2C/blob/744b45a68d4fb5c673d3a5d5b82866a9a940b08a/docker-compose/tests/policies.sh#L19

```bash
...
function exec_docker {
    container_name=$(docker ps -f "name=$DOCKER_NAME_POLICIES" --format '{{.Names}}')
    exec_res=$(docker exec "${container_name}" /bin/bash -c "$1"  2>/dev/null)
}
...
exec_docker "curl -XGET \"${BASE_API_URL}/api/v2/resource-management/policies/leaderinfo\" -ksS"
LEADERINFO=${exec_res}
ISLEADER=$(echo "${LEADERINFO}" | jq -r ".imLeader" 2>/dev/null)
```

Reply:
I have been testing the identification module integration script in the cloud agent, where the full agent is running, and this is the first time I see this error...
Just to clarify, what is the difference between a published port and an exposed one? I ask this because, the port 46060 is exposed by the docker-compose file...

Thank you so much in advance!

Reply:
In the dashboard, there is a container running called identification, outside the current docker-compose network and project, that has the port "46060" exposed to the host:

```
"Ports": {
                "46060/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "46060"
                    }
                ],
                "4848/tcp": null,
                "8080/tcp": null,
                "8181/tcp": null
            },
```

My test, performed inside the mF2C testbed, doesn't has that configuration and consequently fails. That means that it will fail for all the persons that run the test.

Inspect of the deployed mf2c identification

```
            "Ports": {
                "46060/tcp": null,
                "4848/tcp": null,
                "8080/tcp": null,
                "8181/tcp": null
            },
```

and docker ps: 

```
b88563ec37eb        mf2c/identification:v2                        "/bin/sh -c 'asadmin…"   6 minutes ago       Up 6 minutes (healthy)     4848/tcp, 8080/tcp, 8181/tcp, 46060/tcp                        mf2c_identification_1
```

> Just to clarify, what is the difference between a published port and an exposed one? I ask this because, the port 46060 is exposed by the docker-compose file

https://docs.docker.com/compose/compose-file/#expose
https://docs.docker.com/compose/compose-file/#ports
https://nickjanetakis.com/blog/docker-tip-59-difference-between-exposing-and-publishing-ports